### ISSUE TITLE
current avg loss

### ISSUE THREAD

**Issue by abdou31** [AlexeyAB/darknet#2762]
Hello AlexeyAB,
I want to ask about avg loss that increases and decreases during training
Is this normally for training and to precision in the object detection?
( I have one class and i set size of image in cfg file to 64*64) in training 
this picture show that i complete the training with yolo-obj_last.weights
https://i.imgur.com/Zg8xG4e.png

**Comment by drapado** [comment]
It's ok for the loss to go up and down between iterations, although overall it should tend to decrease. Also depends on the number of iterations you have already passed (it should decrease faster in the beginning). 0.16 is a good loss, see this [https://github.com/AlexeyAB/darknet#when-should-i-stop-training](url). However, a better indicator of the performance is the mAP instead of the loss, you can train with -map flag in the end or run the command
./darknet detector map data/obj.data yolo.cfg yolo.weigths

It's everything explained in the link above

**Comment by abdou31** [comment]
Since a few hours, I did the training of model and by approximation, I think that the value of avg loss does not change in the last iterations:
it takes value between 0.0131 and 0.0147
https://i.imgur.com/SkAkLci.png

**Comment by abdou31** [comment]
I get this output when i use darknet.exe detector map data/obj.data yolo.cfg yolo.weigths
https://i.imgur.com/N9T8tsu.png

**Comment by drapado** [comment]
> Since a few hours, I did the training of model and by approximation, I think that the value of avg loss does not change in the last iterations:

Maybe you have reached the lowest loss that you can get with yolo and your dataset.

> I get this output when i use darknet.exe detector map data/obj.data yolo.cfg yolo.weigths

90% of mAP is a very good result. Is this over the training set of over the validation set? If you go to the file obj.data do you have valid=train.txt or valid=valid.txt?

**Comment by abdou31** [comment]
I have this :
+ classes= 1
+ train  = data/obj/train/train.txt
+ valid  = data/obj/test/test.txt
+ names = data/obj.names
+ backup = backup/

**Comment by drapado** [comment]
The darknet map command runs over the valid dataset, in your case test dataset. This means that you have an mAP of 90% in your test set, which is very good

**Comment by abdou31** [comment]
but in some picture , i didn't get good resultat of iris detection , i don't why?

**Comment by drapado** [comment]
Could be due to several causes. Maybe you need more training data, maybe increasing the resolution (width and height) can help. Have a look to this link [https://github.com/AlexeyAB/darknet#how-to-improve-object-detection](https://github.com/AlexeyAB/darknet#how-to-improve-object-detection)
